### ISSUE TITLE
Improve search alert email subject

### ISSUE THREAD

**Issue by saizai** [freelawproject/courtlistener#904]
Suggestion: change email subject from "New hits for your alerts" to "(alert name) alert: (X) new hit(s if plural)".

**Comment by saizai** [comment]
https://github.com/freelawproject/courtlistener/pull/905

**Comment by mlissner** [comment]
Had a look at #905. It's great, but I'm not sure it covers all the bases. Can we discuss a spec here first and then tweak the code to match? 

I think I'd want to cover the four possible subject lines:

1. One alert, one hit
2. One alert, multiple hits
3. Multiple alerts, some with only one hit
4. Multiple alerts, some with multiple hits

I had considered different subjects based on the number of triggered alerts, but I kind of like this proposal, which is simpler, actually: 

> X Alert(s) triggered: "My alert name", "My other alert name"

or:

> X Alert(s) have hits: "My alert name", "My other alert name"

(Anywhere it says `(s)` is an s that is there when pluralized.)

That removes the number of actual hits from the subject line, but I'm not sure there's much value in that number anyway. It's almost always some low number. In my usage, I see that there are hits, I ignore the number, then I skim the message itself, trying to decide if I should click the link to see them on the site.

What do you think?

**Comment by mlissner** [comment]
Elisa, I think this one is relatively easy code-wise, but that it will take you on a journey through a fair bit of code, thus depositing still more of it in your head.

**Comment by albertisfu** [comment]
@mlissner a question regarding the new Alerts subject structure.

I tested the `Alert(s) have hits: "My alert name", "My other alert name"` structure, and it looks good in the preview. The part regarding `X Alert(s) have hits:` is always visible in the email client (tested in Gmail) on both desktop and mobile.


![Image](https://github.com/user-attachments/assets/4f43163a-50e1-4b56-8ff2-9c119e176c30)

![Image](https://github.com/user-attachments/assets/9fa59cea-442b-4df2-90ec-49d5da9d6a70)


However, the subject can become quite lengthy if the alert names are large and the email contains multiple alerts. This is technically possible if a single email matches multiple alerts from a user. This scenario is more likely with weekly and monthly alerts, where multiple alerts are grouped into one email.

According to [RFC2822](http://www.faqs.org/rfcs/rfc2822.html), lines in an email have two limits: they must be no more than 998 characters, and they should ideally be no more than 78 characters.

```
2.1.1. Line Length Limits

   There are two limits that this standard places on the number of
   characters in a line. Each line of characters MUST be no more than
   998 characters, and SHOULD be no more than 78 characters, excluding
   the CRLF.

   The 998 character limit is due to limitations in many implementations
   which send, receive, or store Internet Message Format messages that
   simply cannot handle more than 998 characters on a line. Receiving
   implementations would do well to handle an arbitrarily large number
   of characters in a line for robustness sake. However, there are so
   many implementations which (in compliance with the transport
   requirements of [[RFC2821](http://www.faqs.org/rfcs/rfc2821.html)]) do not accept messages containing more
   than 1000 character including the CR and LF per line, it is important
   for implementations not to create such messages.
```

This line limit also applies to subjects. If a subject is long, it is recommended to split it into multiple lines.

```
2.2.3. Long Header Fields

   Each header field is logically a single line of characters comprising
   the field name, the colon, and the field body.  For convenience
   however, and to deal with the 998/78 character limitations per line,
   the field body portion of a header field can be split into a multiple
   line representation; this is called "folding".  The general rule is

   that wherever this standard allows for folding white space (not
   simply WSP characters), a CRLF may be inserted before any WSP.  For
   example, the header field:

           Subject: This is a test

   can be represented as:

           Subject: This
            is a test
```


I did various tests on lengthy subjects:

**1168 chars subject:**
![Image](https://github.com/user-attachments/assets/51eab781-87f3-43c4-94c2-f8f4515deb5a)

In Gmail only the first 935 chars are displayed when you open the email.

If you inspect the raw email it's possible to see that the whole subject is there.
![Image](https://github.com/user-attachments/assets/3f614285-ba04-4adc-9b01-18050d1f3c45)


Then I tried to force the subject to fold into multiple lines to see if Gmail would display the entire subject when split into multiple lines:
```
subject_header = Header(
        lengthy_subject,
        charset="utf-8",
        header_name="Subject",
        maxlinelen=78
    )
```

However the result was almost the same, the whole subject is not displayed by the client:

![Image](https://github.com/user-attachments/assets/78779ad9-ac18-4c44-bf0a-05f77c684a23)

![Image](https://github.com/user-attachments/assets/d185b267-2882-425d-ba48-a5864b843f89)


Header documentation mention:

> The maximum line length can be specified explicitly via maxlinelen. For
>         splitting the first line to a shorter value (to account for the field
>         header which isn't included in s, e.g. `Subject') pass in the name of
>         the field in header_name.  The default maxlinelen is 78 as recommended
>         by RFC 2822.

So the line split only applies for the first line.

In short, it seems we can't easily exceed the subject limit because, even if we manage to split the entire subject into multiple 78-character lines, some email clients may not display them fully.

Therefore, it might be best to keep subjects below `998` characters (or, better below Gmail’s `935` character limit) to avoid overly long headers that might not be displayed or worse, could trigger spam filters.

Do you have any suggestions for a good character limit we can use to truncate subjects?

**Comment by mlissner** [comment]
Yeah, let's just use `trunc` and go with Gmail's line length with an ellipsis. Seems simple enough and like the best we can do. No point in making it really long anyway: I don't think people will read past the first bit of the subject.
